### PR TITLE
TASK: Refactor absolute namesace in neos cr / security

### PR DESCRIPTION
**Pull request by johannessteu**

### CODE REVIEW

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/AbstractNodePropertyPrivilege.php [hunk 1]
```diff
@@ -47,7 +47,7 @@
     public function matchesSubject(PrivilegeSubjectInterface $subject)
     {
         if ($subject instanceof PropertyAwareNodePrivilegeSubject === false && $subject instanceof MethodPrivilegeSubject === false) {
-            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\ReadNodePropertyPrivilege" only support subjects of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\PropertyAwareNodePrivilegeSubject" or "TYPO3\Flow\Security\Method\MethodPrivilegeSubject", but we got a subject of type: "%s".', get_class($subject)), 1417018448);
+            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "'.ReadNodePropertyPrivilege::class.'" only support subjects of type "'.PropertyAwareNodePrivilegeSubject::class.'" or "'.MethodPrivilegeSubject::class.'", but we got a subject of type: "%s".', get_class($subject)), 1417018448);
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/AbstractNodePropertyPrivilege.php]: As this is already using sprintf we should make the classes arguments now or at least correctly add spaces between the concatenation.

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/AbstractNodePrivilege.php [hunk 2]
```diff
@@ -96,7 +97,7 @@ public function getCacheEntryIdentifier()
     public function matchesSubject(PrivilegeSubjectInterface $subject)
     {
         if ($subject instanceof NodePrivilegeSubject === false && $subject instanceof MethodPrivilegeSubject === false) {
-            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\AbstractNodePrivilege" only support subjects of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\NodePrivilegeSubject" or "TYPO3\Flow\Security\Method\MethodPrivilegeSubject", but we got a subject of type: "%s".', get_class($subject)), 1417014368);
+            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type '.AbstractNodePrivilege::class.' only support subjects of type '.NodePrivilegeSubject::class.' or '.MethodPrivilegeSubject::class.', but we got a subject of type: "%s".', get_class($subject)), 1417014368);
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/AbstractNodePrivilege.php]: Same problem as further down...

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/RemoveNodePrivilege.php [hunk 3]
```diff
@@ -29,7 +29,7 @@ class RemoveNodePrivilege extends AbstractNodePrivilege
     public function matchesSubject(PrivilegeSubjectInterface $subject)
     {
         if ($subject instanceof NodePrivilegeSubject === false && $subject instanceof MethodPrivilegeSubject === false) {
-            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\RemoveNodePrivilege" only support subjects of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\NodePrivilegeSubject" or "TYPO3\Flow\Security\Method\MethodPrivilegeSubject", but we got a subject of type: "%s".', get_class($subject)), 1417017296);
+            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "'.RemoveNodePrivilege::class.'" only support subjects of type "'.NodePrivilegeSubject::class.'" or "'.MethodPrivilegeSubject::class.'", but we got a subject of type: "%s".', get_class($subject)), 1417017296);
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/RemoveNodePrivilege.php]: Again, sprintf or spaces in concat

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/ReadNodePropertyPrivilege.php [hunk 4]
```diff
@@ -36,6 +37,6 @@ class ReadNodePropertyPrivilege extends AbstractNodePropertyPrivilege
      */
     protected function buildMethodPrivilegeMatcher()
     {
-        return 'within(TYPO3\TYPO3CR\Domain\Model\NodeInterface) && method(.*->(getProperty|getName|isHidden|getHiddenBeforeDateTime|getHiddenAfterDateTime|isHiddenInIndex|getAccessRoles)())';
+        return 'within('.NodeInterface::class.') && method(.*->(getProperty|getName|isHidden|getHiddenBeforeDateTime|getHiddenAfterDateTime|isHiddenInIndex|getAccessRoles)())';
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/ReadNodePropertyPrivilege.php]: spaces in concat.

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/EditNodePropertyPrivilege.php [hunk 5]
```diff
@@ -33,6 +34,6 @@ class EditNodePropertyPrivilege extends AbstractNodePropertyPrivilege
      */
     protected function buildMethodPrivilegeMatcher()
     {
-        return 'within(TYPO3\TYPO3CR\Domain\Model\NodeInterface) && method(.*->(setProperty|setName|setHidden|setHiddenBeforeDateTime|setHiddenAfterDateTime|setHiddenInIndex|setAccessRoles)())';
+        return 'within('.NodeInterface::class.') && method(.*->(setProperty|setName|setHidden|setHiddenBeforeDateTime|setHiddenAfterDateTime|setHiddenInIndex|setAccessRoles)())';
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/EditNodePropertyPrivilege.php]: spaces

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/EditNodePrivilege.php [hunk 6]
```diff
@@ -55,6 +55,6 @@ public function matchesSubject(PrivilegeSubjectInterface $subject)
      */
     protected function buildMethodPrivilegeMatcher()
     {
-        return 'within(TYPO3\TYPO3CR\Domain\Model\NodeInterface) && method(public .*->(?!setIndex)(set|unset|remove)[A-Z].*())';
+        return 'within('.NodeInterface::class.') && method(public .*->(?!setIndex)(set|unset|remove)[A-Z].*())';
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/EditNodePrivilege.php]: spaces

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilegeContext.php [hunk 7]
```diff
@@ -24,7 +24,7 @@ class CreateNodePrivilegeContext extends NodePrivilegeContext
     protected $creationNodeTypes;
 
     /**
-     * @param string|array $creationNodeTypes either an array of supported node type identifiers or a single node type identifier (for example "TYPO3.Neos:Document")
+     * @param string|array $creationNodeTypes either an array of supported node type identifiers or a single node type identifier (for example "Neos.Neos:Document")
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilegeContext.php]: This is not correct yet and should only be changed when the naming is actually changed in master only.

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilege.php [hunk 8]
```diff
@@ -82,6 +82,6 @@ public function getCreationNodeTypes()
      */
     protected function buildMethodPrivilegeMatcher()
     {
-        return 'within(TYPO3\TYPO3CR\Domain\Model\NodeInterface) && method(.*->(createNode|createNodeFromTemplate)())';
+        return 'within('.NodeInterface::class.') && method(.*->(createNode|createNodeFromTemplate)())';
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilege.php]: spaces in concat.

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilege.php [hunk 9]
```diff
@@ -39,7 +39,7 @@ class CreateNodePrivilege extends AbstractNodePrivilege
     public function matchesSubject(PrivilegeSubjectInterface $subject)
     {
         if ($subject instanceof CreateNodePrivilegeSubject === false && $subject instanceof MethodPrivilegeSubject === false) {
-            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\CreateNodePrivilege" only support subjects of type "TYPO3\TYPO3CR\Security\Authorization\Privilege\Node\CreateNodePrivilegeSubject" or "TYPO3\Flow\Security\Method\MethodPrivilegeSubject", but we got a subject of type: "%s".', get_class($subject)), 1417014353);
+            throw new InvalidPrivilegeTypeException(sprintf('Privileges of type "'.CreateNodePrivilege::class.'" only support subjects of type "'.CreateNodePrivilegeSubject::class.'" or "'.MethodPrivilegeSubject::class.'", but we got a subject of type: "%s".', get_class($subject)), 1417014353);
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Security/Authorization/Privilege/Node/CreateNodePrivilege.php]: spaces in concat or sprintf

**Comment by johannessteu** [comment]
fixed

**Comment by kitsunet** [comment]
Fixed last remaining whitespace issues, will merge after travis is done.

**Pull request merged**